app/controllers/UpdateRelationshipController.scala

```diff
@@ -322,8 +324,8 @@ class UpdateRelationshipController @Inject()(
           case _: CannotUpdateRelationship => handle(Logger.warn, InternalServerError(views.html.errors.try_later()))
           case _: CitizenNotFound => handle(Logger.warn, InternalServerError(views.html.errors.citizen_not_found()))
           case _: BadFetchRequest => handle(Logger.warn, InternalServerError(views.html.errors.bad_request()))
-          case _: TransferorNotFound => handle(Logger.warn, InternalServerError(views.html.errors.transferor_not_found()))
-          case _: RecipientNotFound => handle(Logger.warn, InternalServerError(views.html.errors.recipient_not_found()))
+          case _: TransferorNotFound => handle(Logger.warn, NotFound(views.html.errors.transferor_not_found()))
+          case _: RecipientNotFound => handle(Logger.warn, NotFound(views.html.errors.recipient_not_found()))
```

Comment:
Im not sure NotFound is appropriate here as its not a missing resource. Lets get together and discuss this.